packages/react/ds/src/header/components/header-slot.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from 'react';
+import ReactDOMClient from 'react-dom/client';
+import { Icon } from '../../icon/icon.js';
+
+type SloTProps = {
+  item: {
+    slot: React.ReactNode;
+    label?: string;
+    icon?: string;
+  };
+  index: number;
+};
+
+export const addNewGovieHeaderSlotElement = (
+  index: number,
+  { slot }: { slot: React.ReactNode },
+) => {
+  const parent = document.querySelector('#GovieHeader');
+
+  if (parent) {
+    const childNode = document.createElement('div');
+    childNode.id = `SlotContainer-${index}`;
+    childNode.dataset.index = index.toString();
+    childNode.className =
+      'gi-hidden gi-bg-gray-50 gi-px-8 gi-pt-8 gi-pb-14 gi-border-b-2xl gi-border-b-emerald-800';
+
+    parent.append(childNode);
+    const root = ReactDOMClient.createRoot(childNode);
+    root.render(slot);
+
+    return () => {
+      root.unmount();
+      childNode.remove();
+    };
+  }
+};
+
+const SlotContainer = ({ index, item }: SloTProps) => {
+  useEffect(() => {
+    const cleanup = addNewGovieHeaderSlotElement(index, item);
+    return cleanup;
+  }, [index, item]);
+
+  return null;
+};
+
+const SlotItemAction = ({ item: { label, icon }, index }: SloTProps) => {
+  return (
+    <label
+      htmlFor={`ItemActionTrigger-${index}`}
+      className="gi-header-tool-item"
+    >
+      <input
```

Comment:
is it a good idea to use the `Checkbox` component here ?

Reply:
If you are using the `checkbox` why do you need to attach the listener to the buttons with `attachEventsToItemActionTriggers`?

Reply:
I needed to attach events on the buttons to toggle the icons, hide all slots opened (including search) and show the current one.
As we have differents custom items slots it weren't possible to have css rules for that (like SearchTrigger), so attaching the event `change` will allow the all checkboxs close and open just the one clicked.